pkg/asset/machines/azure/machines.go

```diff
@@ -78,12 +80,28 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	return machines, nil
 }
 
-func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string, userDataSecret string, clusterID string, role string, azIdx *int, hyperVGen string) (*machineapi.AzureMachineProviderSpec, error) {
+func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string, userDataSecret string, clusterID string, role string, azIdx *int, capabilities map[string]string) (*machineapi.AzureMachineProviderSpec, error) {
 	var az *string
 	if len(mpool.Zones) > 0 && azIdx != nil {
 		az = &mpool.Zones[*azIdx]
 	}
 
+	hyperVGen, err := getHyperVGenerationVersion(capabilities)
+	if err != nil {
+		return nil, err
+	}
+
+	if mpool.VMNetworkingType == "" {
+		acceleratedNetworking, err := getVMNetworkingCapability(capabilities)
+		if err != nil {
+			logrus.Info(err)
```

Comment:
This error should be returned, not logged.

Reply:
Maybe we should discuss the behavior here. This is getting a bit confusing. This means the capabilities did not include the key for the capability. This was happening on Azure stack hub. The issue here is if they didn't put any value in to start (vmnetworkingtype = ""), I'm not sure we want to return error here. Maybe Azure stack doesn't support it and doesn't have it in the capabilities. This logrus was added from a previous discussion.

Reply:
Hopefully this comment will clarify: https://github.com/openshift/installer/pull/5846/files#r868360255

To me it doesn't seem like getVMNetworkingCapability should return an error at all.

Reply:
>  I'm not sure we want to return error here.

Yeah I agree with this.